include/visibility.h

```diff
@@ -13,18 +13,33 @@
 
 #if defined _WIN32 || defined __CYGWIN__
     #ifdef BUILDING_PHPCPP
-        #ifdef __GNUC__
-            #define PHPCPP_EXPORT __attribute__ ((dllexport))
-        #else
-            #define PHPCPP_EXPORT __declspec(dllexport) // Note: actually gcc seems to also supports this syntax.
+        #ifdef BUILDING_SHARED
+            #ifdef __GNUC__
+                #define PHPCPP_EXPORT __attribute__ ((dllexport))
+            #else
+                #define PHPCPP_EXPORT __declspec(dllexport) // Note: actually gcc seems to also supports this syntax.
+            #endif
+        #else /* build static library */
+          #define PHPCPP_EXPORT
         #endif
+        #define DLL_EXPORT
```

Comment:
Why does it need two different specifiers? Either PHPCPP_EXPORT targets DLL or static build.

Thanks.